### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by FinalAngel**
* Added linting to packages
* Added test framework
* Added formatted documentation
* Aligned code with other addons

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# -*- coding: utf-8 -*-
```
Comment (GaretJax) [inline tests/__init__.py]: No more coding stuff please. Py3 is all utf8 anyway.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+indent_style = space
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+max_line_length = 80
+
+[*.py]
+max_line_length = 120
```
Comment (GaretJax) [inline .editorconfig]: 79, not 120.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+quote_type = single
```
Comment (GaretJax) [inline .editorconfig]: black uses double quotes.

File: docs/conf.py [hunk 4]
```diff
@@ -1,34 +1,34 @@
-#!/usr/bin/env python3
-# -*- coding: utf-8 -*-
+# Configuration file for the Sphinx documentation builder.
```
Comment (GaretJax) [inline docs/conf.py:1]: Run black on this file.

File: tests/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,6 @@
--r requirements-test.txt
+pytest
+pytest-cov
+pytest-pep8
 
 # Automation
 Fabric
```
Comment (GaretJax) [inline tests/requirements.txt]: Not sure if we're still using the fabfile. For sure not for tests.

File: tests/requirements.txt [hunk 6]
```diff
@@ -1,4 +1,6 @@
--r requirements-test.txt
+pytest
```
Comment (GaretJax) [inline tests/requirements.txt]: Would be nice to use tox instead of a separate requirements.txt file for tests.

**Review by GaretJax — CHANGES_REQUESTED** [review]

File: tests/requirements.txt (hunk 6 above)
Comment (FinalAngel) [reply tests/requirements.txt]: @GaretJax I was thinking of keeping it simple for now, want me to add tox already or is it okay to add later if necessary?

Comment (GaretJax) [reply tests/requirements.txt]: Ok for now. There are no tests anyway.

**Review by GaretJax — APPROVED** [review]

**Pull request merged**